kalite/topic_tools/content_recommendation.py

```diff
@@ -116,50 +117,49 @@ def get_next_recommendations(user):
 # list). "Immediately" means the very next exercise after the most recent one the given user has accessed.
 #
+# This function checks if the exercises returned have been accessed already, and only returns those that
+# have not been.
+#
 # A group is defined as a collection of students within the same facility and group (as defined in models)
 def get_group_recommendations(user):
-    user = ExerciseLog.objects.filter(id__lte=1)[0].user        #random person, can delete after                                  
-    most_recent_exercise = get_most_recent_exercises(user)[0]   #get most recently accessed exercise
-    user_list = get_users_in_group('Student', user.group, user.facility) #may need to debug
+ 
+    recent_exercises = get_most_recent_exercises(user)
     
-    counts = [{'temp':0}]  #array of dictionaries to keep track of counts of subsequent exercises
+    user_list = FacilityUser.objects.filter(group=user.group)
 
-    '''     NEEDS DEBUGGING     '''
-    for student in user_list:
-        student_exercises = get_most_recent_exercises(student)
-        
-        #if this student has taken/attempted this exercise
-        if most_recent_exercise in student_exercises and len(student_exercises) > 0:
-            next = student_exercises[0] #start at the most recent one - this will act like a prev pointer
-
-            #loop through all exercises, keeping track of previous 
-            for exercise in student_exercises:
-              
-                #a match, and not the first one (which would imply that there is no next)
-                if(exercise == most_recent_exercise and exercise != next):
-                    
-                    found = False #boolean flag
-                    
-                    for c in counts:
-                        #if a match
-                        if next.exercise_id in c:
-                            c[next.exercise_id] += 1
-                            found = True
+    if recent_exercises:
 
-                    #if exercise not found, then make a new object with it with count = 1 
-                    if not found:
-                        counts.append({next.exercise_id : 1})
+        #If the user has recently engaged with exercises, use these exercises as the basis for filtering
+        user_exercises = ExerciseLog.objects.filter(user__in=user_list).order_by("-latest_activity_timestamp").extra(select={'null_complete': "completion_timestamp is null"}, order_by=["-null_complete", "-completion_timestamp"])
+    
+        exercise_counts = {}
 
-                    #break #stops inner for loop logic
+        for user in user_list:
+            user_logs = user_exercises.filter(user=user)
+            for i, log in enumerate(user_logs):
+                if i > 0:
+                    prev_log = user_logs[i-1]
+                    if log.exercise_id in recent_exercises:
+                        if prev_log.exercise_id in exercise_counts:
```

Comment:
I believe you can remove this `if` statement by initializing `exercise_counts` as a `defaultdict` where the default value is 0.